packages/formstr-app/src/containers/ResponsesNew/components/ResponseDetailModal.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Descriptions, Typography, Button, Space } from 'antd';
+import { Event, nip19, nip44 } from 'nostr-tools';
+import { Tag, Field } from '../../../nostr/types'; 
+
+const { Text } = Typography;
+
+type ResponseDetailItem = {
+  key: string; 
+  question: string;
+  answer: string;
+};
+interface ResponseDetailModalProps {
+  isVisible: boolean;
+  onClose: () => void;
+  event: Event | null; 
+  formSpec: Tag[] | null | undefined; 
+  editKey: string | undefined | null; 
+}
+export const ResponseDetailModal: React.FC<ResponseDetailModalProps> = ({
+  isVisible,
+  onClose,
+  event,
+  formSpec,
+  editKey,
+}) => {
+  const [processedData, setProcessedData] = useState<ResponseDetailItem[]>([]);
+  const [metaData, setMetaData] = useState<{ author?: string, timestamp?: string }>({});
+
+  useEffect(() => {
+    if (isVisible && event) {
+      const authorNpub = nip19.npubEncode(event.pubkey);
+      const timestamp = new Date(event.created_at * 1000).toLocaleString();
+      setMetaData({ author: authorNpub, timestamp });
+      setProcessedData([{ key: 'loading', question: 'Processing...', answer: '...' }]); 
+    } else {
+      setProcessedData([]);
+      setMetaData({});
+    }
+  }, [isVisible, event, formSpec, editKey]); 
+  const getInputs = (responseEvent: Event): Tag[] => {
+     if (responseEvent.content === "") {
+       return responseEvent.tags.filter((tag): tag is Tag => Array.isArray(tag) && tag[0] === "response");
+     } else if (editKey) {
+       try {
+         let conversationKey = nip44.v2.utils.getConversationKey(
+           editKey,
+           responseEvent.pubkey
+         );
+         let decryptedContent = nip44.v2.decrypt(
+           responseEvent.content,
+           conversationKey
+         );
+         const parsed = JSON.parse(decryptedContent);
+         if(Array.isArray(parsed)) {
+             return parsed.filter(
+               (tag: Tag): tag is Tag => Array.isArray(tag) && tag[0] === "response"
+             );
+         }
+         console.warn("Decrypted content is not an array:", parsed);
+         return [];
+       } catch (e) {
+           console.error("Failed to parse decrypted response content in modal:", e);
+           return [];
+       }
+     } else {
+       console.warn("Cannot decrypt response in modal: EditKey not available.");
+       return [];
+     }
+   };
+
+  const processEventForDisplay = (
+    responseEvent: Event,
+    spec: Tag[] | null | undefined,
+    key: string | undefined | null
+  ): ResponseDetailItem[] => {
+    if (!spec) {
```

Comment:
instead of handling all these errors let's just make it mandatory for spec to be preent and not make the modal clickable until we have a formspec? would reduce clutter